adding support for docker stack config

thosil opened the pull request:
The `docker stack config` command returns a yaml content on stdout. We have to parse the yaml output, hence the new pyyaml requirement.

The parameter "return_json" like the `ComposeCLI.config` method.

+ a couple of tests

fix: #509

thosil committed:
adding support for docker stack config

The `docker stack config` command returns a yaml content on stdout.
We have to parse the yaml output, hence the new pyyaml requirement.

The parameter "return_json" like the `ComposeCLI.config` method.

+ a couple of tests

thosil committed:
fix ci

- removed unused import
- support pydantic 1.9

thosil committed:
fix ci

- reformat files

gabrieldemarmiesse commented on requirements.txt:
```diff
@@ -3,3 +3,4 @@ requests
 tqdm
 typer>=0.4.1
 typing_extensions
+pyyaml >= 6.0.1
```
I think this should be an optional dependency as only a small percentage of users will actually use `docker.stack.config()`. We don't want to force this dependency on everyone.

thosil replied:
ok, pyyaml is optional in the latest commit and could be installed as extra (pip install python_on_whales[yaml])
I also had to try importing the yaml module and handle the fact it could be missing in the config() method.

I don't know if it's the better way to do it.

thosil committed:
make pyyaml optional

- moved pyyaml dep from requirements.txt to pyproject.toml (user will have to install python_on_whales[yaml] to have it)
- handle missing dependency in stack/cli_wrapper.py
- add pyyaml in test-requirements.txt

thosil commented:
:grimacing: I forgot to add a docstring and I don't have the time to do it for the moment, sorry

gabrieldemarmiesse commented:
No worries, we're not in a hurry. In the open source world, we have all the time we want

thosil committed:
add documentation

thosil committed:
Apply suggestions from code review

Co-authored-by: Gabriel de Marmiesse <gabrieldemarmiesse@gmail.com>

thosil committed:
Merge branch 'master' into add_support_for_stack_config

thosil committed:
Merge branch 'master' into add_support_for_stack_config

gabrieldemarmiesse commented:
Closing as stale. Feel free to open another PR if you want!
